staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -720,6 +730,137 @@ func ValidateCustomResourceColumnDefinition(col *apiextensions.CustomResourceCol
 	return allErrs
 }
 
+// validateFieldPath validates a field path against the given schema
+func validateFieldPath(fieldPath string, pathOfFieldPath *field.Path, schema *apiextensions.JSONSchemaProps) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(fieldPath) == 0 {
+		allErrs = append(allErrs, field.Invalid(pathOfFieldPath, fieldPath, "must not be empty"))
+		return allErrs
+	}
+
+	nonexistentFieldError := field.Invalid(pathOfFieldPath, fieldPath, "does not refer to an existent field")
+
+	var s scanner.Scanner
```

Comment:
is the grammar recognized by `text/scanner` compatible with jsonpath? I was surprised to see this used in this way

Reply:
I had suggested using this as a quick way to parse a subset of JSON path for this use case.

I forgot to check if the identifier formats are compatible, everything else should be safe.

Reply:
scanner here is just used to simple check if the field name exists in the schema. I thought of ways for API validation:
1. using cel for path check. pro: safe and accurate. con: extra cost, might be abused.
2. simple check like other jsonpath field in CRD which did nothing..
3. check the basic field name existence using scanner which is the current impl
Please feel free to advice on concerns or other preferred direction. Thank you

Reply:
To capture our offline conversation. If we use a minimal subset of JSON Path and ensure we're using it in a safe way that also produces canonical field path output in validation error messages we should be safe. Specifically:

1. remove numeric index support for list (since hard coded values have no known uses)
 2. only support dot-notation for properties and array-notation values with strings for additionalProperties
 3. Ensure field name compatibility between CRD allowed field names and our mini-grammar (unicode char support, ...)
 4. Use single quoted for strings like shown in all JSON Path examples, with compatible escaping, so that we are complaint with JSON Path (the lack of a proper spec for JSON Path makes this difficult, but we're using a small subset)

Reply:
Have updated. In consideration of removing ambiguity, map key(child operator ['specialName']) would be used for field name which has special character. e.g. for a field named `a.b` under list, it will be referred as `.list['a.b']`.